tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -13236,6 +13205,173 @@ void ExecutionTest::WaveSizeTest() {
+                             Data.size());
+              WaveSizeTestData *pInData = (WaveSizeTestData *)Data.data();
+              memset(pInData, 0, sizeof(WaveSizeTestData) * MAX_WAVESIZE);
+            },
+            ShaderOpSet);
+
+        // verify expected values
+        MappedData dataUav;
+        WaveSizeTestData *pOutData;
+
+        LogCommentFmt(
+            L"Verifying test result for these shader wave size ranges:");
+        LogCommentFmt(L"Minimum shader wave size: %d", minShaderWaveSize);
+        LogCommentFmt(L"Maximum shader wave size: %d", maxShaderWaveSize);
+        LogCommentFmt(L"Preferred shader wave size: %d", prefShaderWaveSize);
+
+        // at this point we assume that the waverange size that
+        // the shader specifies is legal.
+        test->Test->GetReadBackData("UAVBuffer0", &dataUav);
+        VERIFY_ARE_EQUAL(sizeof(WaveSizeTestData) * MAX_WAVESIZE,
+                         dataUav.size());
+        pOutData = (WaveSizeTestData *)dataUav.data();
+
+        for (unsigned i = 0; i < MAX_WAVESIZE; i++) {
+          VERIFY_IS_TRUE(pOutData[i].count >= minWaveSize &&
+                         pOutData[i].count <= maxWaveSize);
+
+          if (prefShaderWaveSize >= minWaveSize &&
+              prefShaderWaveSize <= maxWaveSize) {
+
+            VERIFY_ARE_EQUAL(pOutData[i].count, prefShaderWaveSize);
+          }
+        }
+      }
+    }
+  }
+}
+
+void ExecutionTest::WaveSizeTest() {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  CComPtr<ID3D12Device> pDevice;
+  if (!CreateDevice(&pDevice, D3D_SHADER_MODEL_6_6)) {
+    return;
+  }
+
+  // Check Wave support
+  if (!DoesDeviceSupportWaveOps(pDevice)) {
+    // Optional feature, so it's correct to not support it if declared as such.
+    WEX::Logging::Log::Comment(L"Device does not support wave operations.");
+    return;
+  }
+
+  // Get supported wave sizes
+  D3D12_FEATURE_DATA_D3D12_OPTIONS1 waveOpts;
+  VERIFY_SUCCEEDED(
+      pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS1,
+                                   &waveOpts, sizeof(waveOpts)));
+  UINT minWaveSize = waveOpts.WaveLaneCountMin;
+  UINT maxWaveSize = waveOpts.WaveLaneCountMax;
+
+  DXASSERT_NOMSG(minWaveSize <= maxWaveSize);
+  DXASSERT((minWaveSize & (minWaveSize - 1)) == 0, "must be a power of 2");
+  DXASSERT((maxWaveSize & (maxWaveSize - 1)) == 0, "must be a power of 2");
+
+  // read shader config
+  CComPtr<IStream> pStream;
+  std::shared_ptr<st::ShaderOpSet> ShaderOpSet =
+      std::make_shared<st::ShaderOpSet>();
+  ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
+  st::ParseShaderOpSetFromStream(pStream, ShaderOpSet.get());
+
+  LogCommentFmt(L"Testing WaveSize attribute for shader model 6.6.");
+  RunWaveSizeTest(minWaveSize, maxWaveSize, ShaderOpSet, pDevice, m_support);
+  pDevice.Release();
+  if (!CreateDevice(&pDevice, D3D_SHADER_MODEL_6_8)) {
```

Comment:
For this case, if we fail to create SM 6.8 device, we don't want to report skip for the overall result of this test, so we need to pass false to the `skipUnsupported` parameter, like:
```cpp
  if (!CreateDevice(&pDevice, D3D_SHADER_MODEL_6_8, /*skipUnsupported*/ false)) {
```